File: program/find-sum-of-digits-of-a-number-by-recursion/find_sum_of_digits_of_a_number_by_recursion.rs
use std::io;
use std::io::Write;

fn main() {    
    let mut num = String::new();
    let mut sum: i64 = 0;
    
    print!("Insert number: ");
    io::stdout().flush().unwrap();
    
    io::stdin()
        .read_line(&mut num)
        .expect("Failed to read line");
    
    let mut num: i64 = num
        .trim()
        .parse()
        .expect("Please enter a valid number!");
    
    while num > 0 {
        sum += num%10;
        num /= 10
    }
    println!("{sum}");
}

File: program/find-sum-of-digits-of-a-number-by-recursion/FindSumOfDigitsOfANumberByRecursion.rb
def SumOfDigits(num,sum)
	if num > 0 
		sum += (num % 10);
		SumOfDigits(num / 10,sum);
	else
	    return sum;    
	end
end

print "Enter a number: ";
number = gets.chomp.to_i;  

result = SumOfDigits(number, 0);
print "Sum of digits: ",result; 
